src/main.rs

```diff
@@ -466,7 +469,16 @@ impl Future for Main {
             if let Some(ref mut player_event_channel) = self.player_event_channel {
                 if let Async::Ready(Some(event)) = player_event_channel.poll().unwrap() {
                     if let Some(ref program) = self.player_event_program {
-                        run_program_on_events(event, program);
+                        let child = run_program_on_events(event, program)
+                            .expect("program failed to start");
```

Comment:
note: this is bad behavior and should probably be handled some other way than panicking, but it's what the existing code does...